### DISCUSSION TITLE
Support for multilingual and multi-regional storefronts

### DISCUSSION THREAD

**Discussion by egant** [rylanharper/nitrogen#11] [Ideas]
This kicks off a discussion on adding more localization capabilities to Nitrogen.

The [ Shopify Hydrogen docs](https://shopify.dev/docs/storefronts/headless/hydrogen/markets/multiple-languages-url-paths)  outline how to leverage Shopify's Markets with Hydrogen to create shopping experiences in local languages and currencies.

Being able to translate the Nitrogen UI and fetch localized content for product descriptions, blog entries, etc. would be neat.

I want to ask if this is interesting / planned / possible?

**Comment by rylanharper** [comment]
Hey @egant, this is a great question! 

I'm looking into adding [Nuxt I18n](https://i18n.nuxtjs.org/) to configure locale-based route parameters, enabling URLs like `your-site-domain.com/fr/products` that align with the corresponding ISO code. However, this isn't a priority in the short term—at least for the next month or so (I am focusing on Shopify Analytics and Sanity integration).

That said, this is likely as far as I will take localization for a few reasons.. In the vast majority (like 95%) of headless sites, the default language is either English or set to a specific region for simplicity. When language variations are required, they are often managed through third-party solutions like CMS platforms with localization features, specific Shopify apps, or other paid services. Additionally, browsers like Google Chrome can automatically detect a webpage's language and offer translation options. While not perfect, these tools have improved significantly and generally provide a reliable experience.

So to answer your question: I plan to implement basic i18n support, but more advanced language features are outside the scope of this project.

PS - This project is also somewhat more minimal in scope than Hydrogen itself (mainly because its just me haha), but I am trying to add as much support for various features as I can without it becoming too overwhelming.

**Comment by egant** [comment]
Adding Nuxt i18n would be really awesome if / when you can find the time / energy.

As for this project being mainly just you, I think given the quality and feature set of what you have made there is no doubt whatsoever that folks will want to contribute.  I first saw this Nuxt based project https://github.com/scottyzen/woonuxt about 20 months ago and the main dev (Scott) has now completed migration to Nuxt 3 and picked up something like 15 contributors along the way.  So... if you build it they will come!

When you say "outside the scope" would you still be open to reviewing PR's that take supported localization features beyond Nuxt i18n? I am interested in fetching localized product descriptions and blog entries from the Storefront API. The concept of managing everything through a single admin application — one with a low learning curve, built-in RBAC, a free translation management tool, and having the ability to query a single API at build or runtime—appeals to me. This approach not only simplifies workflows but also eliminates the need for additional hosting or CMS subscriptions, reducing operational costs.

**Reply by rylanharper** [reply]
Hey @egant 

Sorry for the late reply on this, I was super busy with work last week. 

Yes, I am open to reviewing any PRs for this project, whether they're about localization or any other improvements! I should hopefully find some time to add in i18n next month. I think the main issue with not having many devs contributing so far is that not many know the ins-and-outs of Shopify development in a Nuxt/Vue environment. One of my key goals with this project is to establish and share best practices for using the Storefront API with Nuxt. Hopefully that encourages more developers to dive in.

PS - I added in pagination a few days ago to collection/search pages so be sure to check that out:)

**Reply by alefbarbeli** [reply]
I would be happy to collab on this nuxt-i18n feature. @rylanharper are you okay with me working on it?

**Reply by rylanharper** [reply]
@alefbarbeli That would be great! This template has come a long way since this discussion started here, but I did not have the time to add in i18n support. Let me start a new issue (labeled with the `enhancement` tag) and we can go over some details regarding this. I also have some ideas.

**Reply by alefbarbeli** [reply]
@rylanharper I'll wait for the new issue so we follow up there!

**Reply by rylanharper** [reply]
@alefbarbeli  Just created it - kind of a WIP, but it should be enough to get an outline going.
